enable testing with podman

mikebonnet opened the pull request:
`podman` allows containers to be run by non-root users, and requires no special privileges or
background daemons. It is available natively on most modern Linux distros, as well as
on Windows and MacOS. Enabling `stomp.py` to be tested using `podman` expands the environments
where `stomp.py` development is possible.

Note that `podman` >= 4.0.0 is required (for static ip support). Tested using `podman` 4.1.0 on
Fedora 36 x86_64.

mikebonnet committed:
enable testing with podman

podman allows containers to be run by non-root users, and requires no special privileges or
background daemons. It is available natively on most modern Linux distros, as well as
on Windows and MacOS. Enabling stomp.py to be tested using podman expands the environments
where stomp.py development is possible.

Note that podman >= 4.0.0 is required (for static ip support). Tested using podman 4.1.0 on
Fedora 36 x86_64.

jasonrbriggs commented:
I get an error when I try to run it...

```
Error: ipv6 option requires an IPv6 --subnet to be provided
```

mikebonnet commented:
Oh interesting, I didn't get that in my testing. What version of `podman`, on what OS?

mikebonnet committed:
set a static IPv6 subnet for the podman network

It appears that cni-based networking does not require this, but netavark does. Setting
static subnets for IPv4 and IPv6 works everywhere.

mikebonnet commented:
I just added a change which sets a static IPv6 address as well. Please give it a try.

mikebonnet commented:
Have you had a chance to retest this?

jasonrbriggs commented:
Sorry I did and then promptly forgot to reply.  I get these errors with the latest update:

```
podman ps
CONTAINER ID  IMAGE                     COMMAND     CREATED       STATUS                     PORTS                                                                                                               NAMES
05345fa29b80  localhost/stomppy:latest              1 second ago  Up Less than a second ago  0.0.0.0:61613->61613/tcp, 0.0.0.0:62613-62614->62613-62614/tcp, 0.0.0.0:63613->63613/tcp, 0.0.0.0:64613->64613/tcp  stomppy
podman exec -it stomppy /bin/sh -c "/etc/init.d/activemq start"
Starting ActiveMQ instance : activemq expiredbroker (running) main (running).
podman exec -it stomppy /bin/sh -c "/etc/init.d/stompserver start"
Starting stompserver : stompserverChanging group to stompserver.
Changing user to stompserver.
 failed!
podman exec -it stomppy /bin/sh -c "/etc/init.d/rabbitmq-server start"
Starting RabbitMQ Messaging Server: rabbitmq-serverFAILED - check /var/log/rabbitmq/startup_\{log, _err\} ... (warning).
 failed!
make: *** [Makefile:107: run-podman] Error 1
```

mikebonnet commented:
Strange, I just tested it here and `make podman; make test` works.

Could you paste the output of `podman info`?

mikebonnet commented:
Oh, your `debian` base image may be out-of-date. Try a `podman pull debian` before running `make podman`.

jasonrbriggs commented:
Same issue even after updating debian.

mikebonnet commented:
Could you paste the output of `podman info` and `podman exec stomppy cat /var/log/rabbitmq/startup_err` ?

jasonrbriggs commented:
podman info:

```
host:
  arch: amd64
  buildahVersion: 1.23.1
  cgroupControllers:
  - memory
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: 'conmon: /usr/bin/conmon'
    path: /usr/bin/conmon
    version: 'conmon version 2.0.25, commit: unknown'
  cpus: 4
  distribution:
    codename: jammy
    distribution: ubuntu
    version: "22.04"
  eventLogger: journald
  hostname: jasonrbriggs-laptop
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 5.15.0-39-generic
  linkmode: dynamic
  logDriver: journald
  memFree: 4995694592
  memTotal: 8196816896
  ociRuntime:
    name: runc
    package: 'runc: /usr/sbin/runc'
    path: /usr/sbin/runc
    version: |-
      runc version 1.1.0-0ubuntu1
      spec: 1.0.2-dev
      go: go1.17.3
      libseccomp: 2.5.3
  os: linux
  remoteSocket:
    exists: true
    path: /run/user/1000/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: 'slirp4netns: /usr/bin/slirp4netns'
    version: |-
      slirp4netns version 1.0.1
      commit: 6a7b16babc95b6a3056b33fb45b74a6f62262dd4
      libslirp: 4.6.1
  swapFree: 2147479552
  swapTotal: 2147479552
  uptime: 3m 56.77s
plugins:
  log:
  - k8s-file
  - none
  - journald
  network:
  - bridge
  - macvlan
  volume:
  - local
registries: {}
store:
  configFile: /home/jasonrbriggs/.config/containers/storage.conf
  containerStore:
    number: 1
    paused: 0
    running: 0
    stopped: 1
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /home/jasonrbriggs/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 37
  runRoot: /run/user/1000/containers
  volumePath: /home/jasonrbriggs/.local/share/containers/storage/volumes
version:
  APIVersion: 3.4.4
  Built: 0
  BuiltTime: Thu Jan  1 01:00:00 1970
  GitCommit: ""
  GoVersion: go1.17.3
  OsArch: linux/amd64
  Version: 3.4.4
```

jasonrbriggs commented:
startup_err:

```
Configuring logger redirection

BOOT FAILED
===========
Error during startup: {error,
                       {rabbitmq_stomp,
                        {{shutdown,
                          {failed_to_start_child,
                           'rabbit_stomp_listener_sup_172.17.0.2:61613',
                           {shutdown,
                            {failed_to_start_child,
                             {ranch_listener_sup,
                              {acceptor,{172,17,0,2},61613}},
                             {shutdown,
                              {failed_to_start_child,ranch_acceptors_sup,
                               {listen_error,
                                {acceptor,{172,17,0,2},61613},
                                eaddrnotavail}}}}}}},
                         {rabbit_stomp,start,[normal,[]]}}}}

init terminating in do_boot ({error,{rabbitmq_stomp,{{shutdown,{_}},{rabbit_stomp,start,[_]}}}})

Crash dump is being written to: /var/log/rabbitmq/erl_crash.dump...done
```

jasonrbriggs commented:
The "eaddrnotavail" error is interesting. If I check my ports with netstat, i can see something is listening on 61613 (couldn't quite see what was running though).  So somewhere in the search/replace stuff it's going wrong perhaps? Why just when using podman and not docker...?

mikebonnet commented:
Ahh, I did my testing on podman 4.1.0, you're using 3.4.4. Any chance you could test with a more recent version of podman?

https://packages.debian.org/experimental/podman

jasonrbriggs commented:
Seems difficult actually.  I'm running ubuntu, and the default version is 3.4.4.  To build from source I need a bunch of deps installed and at least one of those (libsystemd-dev) is a pain to install (due to something dodgy about my system).

jasonrbriggs commented:
I can't get podman to work on my laptop annoyingly.  No reason not to merge this though since it works fine without podman at least...
